analysis/demonhuntervengeance/src/modules/shadowlands/legendaries/BlindFaith.tsx

```diff
@@ -0,0 +1,78 @@
+import SPELLS from 'common/SPELLS';
+import COVENANTS from 'game/shadowlands/COVENANTS';
+import Analyzer, { Options, SELECTED_PLAYER } from 'parser/core/Analyzer';
+import Events, { EventType } from 'parser/core/Events';
+import BoringSpellValueText from 'parser/ui/BoringSpellValueText';
+import Statistic from 'parser/ui/Statistic';
+import STATISTIC_CATEGORY from 'parser/ui/STATISTIC_CATEGORY';
+
+import { ConsumeSoulFragmentsEvent } from '../../statistics/SoulFragmentsConsume';
+
+/**
+ * Blind Faith
+ * Shadowlands legendary that buffs the player after casting Elysian Decree
+ * for 20 seconds
+ */
+export default class BlindFaith extends Analyzer {
+  blindFaithActive = false;
+  currentStacks = 0;
+  stacksGained: number[] = [];
+
+  constructor(options: Options) {
+    super(options);
+    this.active =
+      this.selectedCombatant.hasCovenant(COVENANTS.KYRIAN.id) &&
+      this.selectedCombatant.hasLegendary(SPELLS.BLIND_FAITH);
+
+    if (!this.active) {
+      return;
+    }
+
+    this.addEventListener(
+      Events.applybuff.by(SELECTED_PLAYER).spell(SPELLS.BLIND_FAITH_BUFF),
+      this.onBlindFaithGain,
+    );
+    this.addEventListener(
+      Events.removebuff.by(SELECTED_PLAYER).spell(SPELLS.BLIND_FAITH_BUFF),
+      this.onBlindFaithLoss,
+    );
+    this.addEventListener(EventType.ConsumeSoulFragments, this.onSoulFragmentsConsumed);
+  }
+
+  onBlindFaithGain() {
+    this.blindFaithActive = true;
+    // e.g. when recasting Decree while Urh is up, you lose internal stacks
```

Comment:
You might want to add another event listener for refreshbuff then
Since most buffs will not do another apply event if they already have the buff